backend/clickhouse/query.go

```diff
@@ -363,10 +363,15 @@ func KeysAggregated(ctx context.Context, client *Client, tableName string, proje
 		From(tableName).
 		Where(sb.Equal("ProjectId", projectID)).
 		Where(fmt.Sprintf("Day >= toStartOfDay(%s)", sb.Var(startDate))).
-		Where(fmt.Sprintf("Day <= toStartOfDay(%s)", sb.Var(endDate))).
-		GroupBy("1").
+		Where(fmt.Sprintf("Day <= toStartOfDay(%s)", sb.Var(endDate)))
+
+	if query != nil && *query != "" {
+		sb.Where(fmt.Sprintf("Key LIKE %s", sb.Var("%"+*query+"%")))
+	}
+
+	sb.GroupBy("1").
 		OrderBy("2 DESC, 1").
-		Limit(500)
+		Limit(10)
```

Comment:
does the limit actually affect the max number of rows clickhouse scans when building the result set? limiting to the query should help with that tho

Reply:
Gotcha! I 'm seeing that the error is coming from this setting
https://github.com/highlight/highlight/blob/main/backend/clickhouse/query.go#L358

Seems like we wouldn't want it to error here if their were too many rows to scan?